source/ui/SGM/SGMXASScanConfigurationView.cpp

```diff
@@ -246,6 +261,53 @@ void SGMXASScanConfigurationView::setDirection(AMScanConfiguration::Direction ne
 		decreaseRadioButton_->setChecked(true);
 }
 
+void SGMXASScanConfigurationView::updateTimeBounds()
+{
+	if(gratingAngleControl_->isConnected() &&
+	        gratingTranslationStepControl_->isConnected()) {
+
+		SGMGratingSupport::GratingTranslation currentTranslation = SGMGratingSupport::encoderCountToEnum(gratingTranslationStepControl_->value());
+
+		if(currentTranslation != SGMGratingSupport::UnknownGrating) {
+
+			QPair<double, double> timeBounds = gratingAngleControl_->timeBoundsForEnergyMove(axisStart_->value(),
+																							 axisEnd_->value(),
+																							 currentTranslation);
+
+			if(timeBounds.first > 0 && timeBounds.second > 0) {
+
+				if(dwellTime_->value() > timeBounds.second) {
+
+					dwellTime_->setValue(timeBounds.second);
+
+				} else if(dwellTime_->value() < timeBounds.first) {
+
+					dwellTime_->setValue(timeBounds.first);
+				}
+
+				dwellTime_->setRange(timeBounds.first, timeBounds.second);
+
+				timeBoundsLabel_->setText(QString("Valid times for energy range are: Min:%3s  Max:%4s")
+										  .arg(timeBounds.first, 0, 'f', 2)
+										  .arg(timeBounds.second, 0, 'f', 2));
+			}
```

Comment:
It looks like the timeBoundsLabel_ is not updated if the current grating is UnknownGrating or if the time bounds from the grating angle control are invalid. Should the view indicate a problem in this case?